src/Query/QueryExpressionBuilder.php

```
@@ -27,8 +28,12 @@ public function __construct(private QueryBuilderInterface $queryBuilder)
      * @throws InvalidConfigException
      * @throws NotSupportedException
      */
-    public function build(QueryInterface $expression, array &$params = []): string
+    public function build(ExpressionInterface $expression, array &$params = []): string
     {
+        if (!$expression instanceof QueryInterface) {
+            throw new InvalidConfigException('QueryExpressionBuilder can only be used with QueryInterface instance.');
```

Should we throw `InvalidArgumentException` or `InvalidConfigException` here?